core/bin/zksync_server/src/main.rs

```diff
@@ -177,7 +180,13 @@ fn main() -> anyhow::Result<()> {
         return Ok(());
     }
 
-    node.build(opt.components.0)?.run(observability_guard)?;
+    let built_node = node.build(opt.components.0)?;
+
+    if opt.only_verify_config {
+        return Ok(());
+    }
+
+    built_node.run(observability_guard)?;
```

Comment:
Component wiring with some checks happens in `node::run`. In most cases it should be ok to wire to just check the config but there are layers that read from DB during wiring, e.g. `OutputHandlerLayer`. We can either leave it as is (do not wire), or refactor so wiring doesn't require DB access. I'm not sure what's better in this case, we want to validate configs as much as possible, but OTOH we want as little code as possible to be executed.

Reply:
Some layers [also access L1 during wiring](https://github.com/matter-labs/zksync-era/blob/8e24403190c4f7e306c88ccdb3c2f4a86e0e56fa/core/node/node_framework/src/implementations/layers/pk_signing_eth_client.rs#L76-L79) 🙃 IMO, it could be worth clarifying how we define a correct config; it not trivial whether it should include or exclude stuff like DBs and L1 client being accessible. FWIW, [config validation](https://github.com/matter-labs/zksync-era/blob/88e4ed3bf660ca565fc6f8a556b6965b542bb393/core/bin/zksync_server/src/main.rs#L130-L141) built into [the new config system PoC](https://github.com/matter-labs/zksync-era/compare/aov-pla-1068-configuration-system-poc) just checks that a config is deserializable.